packages/react-dom-bindings/src/server/ReactDOMServerExternalRuntime.js

```diff
@@ -13,13 +14,63 @@ import {
   completeSegment,
 } from './fizz-instruction-set/ReactDOMFizzInstructionSet';
 
-// Intentionally does nothing. Implementation will be added in future PR.
-// eslint-disable-next-line no-unused-vars
-const observer = new MutationObserver(mutations => {
-  // These are only called so I can check what the module output looks like. The
-  // code is unreachable.
-  clientRenderBoundary();
-  completeBoundaryWithStyles();
-  completeBoundary();
-  completeSegment();
-});
+// This runtime may be sent to the client multiple times (if FizzServer.render
+//  is called more than once). Here, we check whether the mutation observer
+//  was already created / installed
+if (!window.$REACT_FIZZ_OBSERVER) {
+  // TODO: Eventually remove, we currently need to set these globals for
+  // compatibility with ReactDOMFizzInstructionSet
+  window.$RC = completeBoundary;
+  window.$RM = new Map();
+  window.$REACT_FIZZ_OBSERVER = new MutationObserver(mutations => {
+    for (let i = 0; i < mutations.length; i++) {
+      const addedNodes = mutations[i].addedNodes;
+      for (let j = 0; j < addedNodes.length; j++) {
+        handleNode(addedNodes.item(j));
+      }
+    }
+  });
+  // $FlowFixMe[incompatible-call] document.body should exist at this point
+  window.$REACT_FIZZ_OBSERVER.observe(document.body, {
```

Comment:
This will keep observing for the life time of the page. We should stop observing after the `DOMContentLoaded` event.

Similarly, if this script doesn't load until the document has loaded we should not observe at all. Meaning we should only add the observer if `document.readyState === 'loading'`.